assets/js/membraneWebRTC.ts

```diff
@@ -362,20 +412,109 @@ export class MembraneWebRTC {
-    const sender = this.connection!.getSenders().find((sender) => {
-      return sender!.track!.id === oldTrackId;
+  public async replaceTrack(
+    oldTrackId: string,
+    newTrack: MediaStreamTrack,
+    newTrackMetadata?: any
+  ): Promise<boolean> {
+    const trackContext = this.getTrackContext(oldTrackId);
+    const sender = this.findSender(trackContext.track.id);
+    if (sender) {
+      return sender
+        .replaceTrack(newTrack)
+        .then(() => {
+          const trackMetadata = newTrackMetadata || this.trackIdToTrack.get(oldTrackId)!.metadata;
+          trackContext.track = newTrack;
+          this.updateTrackMetadata(oldTrackId, trackMetadata);
+          return true;
+        })
+        .catch(() => false);
+    }
+
+    return false;
+  }
+
+  public removeTrack(trackId: string) {
+    const trackContext = this.getTrackContext(trackId);
+    const sender = this.findSender(trackContext.track.id);
+    this.connection!.removeTrack(sender);
+    let mediaEvent = generateMediaEvent("renegotiateTracks", {});
+    this.sendMediaEvent(mediaEvent);
+  }
+
+  private findSender(trackId: string): RTCRtpSender {
+    return this.connection!.getSenders().find(
+      (sender) => sender.track && sender!.track!.id === trackId
+    )!;
+  }
+
+  /**
+   * Updates the metadata for the current peer.
+   * @param peerMetadata - Data about this peer that other peers will receive upon joining.
+   *
+   * If the metadata is different from what is already tracked in the room, the optional
+   * callback `onPeerUpdated` will be triggered for other peers in the room.
+   */
+  public updatePeerMetadata = (peerMetadata: any): void => {
+    let mediaEvent = generateMediaEvent("updatePeerMetadata", {
+      metadata: peerMetadata,
     });
-    return sender!.replaceTrack(newTrack);
+    this.sendMediaEvent(mediaEvent);
+  };
+
+  /**
+   * Updates the metadata for a specific track.
+   * @param track - Audio or video track.
+   * @param trackMetadata - Data about this track that other peers will receive upon joining.
+   *
+   * If the metadata is different from what is already tracked in the room, the optional
+   * callback `onPeerUpdated` will be triggered for other peers in the room.
+   */
+  public updateTrackMetadata = (trackId: string, trackMetadata: any): void => {
+    const trackContext = this.getTrackContext(trackId);
+    trackContext.metadata = trackMetadata;
+    this.trackIdToTrack.set(trackId, trackContext);
+    const peer = this.idToPeer.get(this.id!)!;
+    peer.trackIdToMetadata.set(trackId, trackMetadata);
+    let mediaEvent = generateMediaEvent("updateTrackMetadata", {
+      trackId,
+      trackMetadata,
+    });
+    this.sendMediaEvent(mediaEvent);
+  };
+
+  private getTrackContext(trackId: string): TrackContext {
+    const trackContext = this.trackIdToTrack.get(trackId);
+    if (!trackContext) throw "Track with this id doesn't exist";
+    return trackContext;
   }
```

Comment:
I am not sure if we need this but if we want to keep this then we need to print `id` of track in error message 
```js
if (!trackContext) throw "Track with id: xxxx doesn't exist";
```